swf/src/types/fixed.rs

```diff
@@ -0,0 +1,643 @@
+                let n = (<$intermediate_type>::from(self.0) * <$intermediate_type>::from(other))
+                    >> Self::FRACTIONAL_BITS;
+                if cfg!(debug_assertions) {
+                    // Check for overflow.
+                    <$underlying_type>::try_from(n).expect("Attempted to multiply with overflow")
+                } else {
+                    n as $underlying_type
+                }
+            }
+
+            /// Wrapping (modular) negation. Computes -self, wrapping around at the boundary of the type.
+            /// -Self::MIN is the only case where wrapping occurs.
+            #[inline]
+            pub fn wrapping_neg(self) -> Self {
+                Self(self.0.wrapping_neg())
+            }
+
+            /// Wrapping (modular) addition. Computes self + rhs, wrapping around at the boundary of the type.
+            #[inline]
+            pub fn wrapping_add(self, other: Self) -> Self {
+                Self(self.0.wrapping_add(other.0))
+            }
+
+            /// Wrapping (modular) subtraction. Computes self - rhs, wrapping around at the boundary of the type.
+            #[inline]
+            pub fn wrapping_sub(self, other: Self) -> Self {
+                Self(self.0.wrapping_sub(other.0))
+            }
+
+            /// Wrapping (modular) multiplication. Computes self * rhs, wrapping around at the boundary of the type.
+            #[inline]
+            pub fn wrapping_mul(self, other: Self) -> Self {
+                let n = <$intermediate_type>::from(self.0)
+                    .wrapping_mul(<$intermediate_type>::from(other.0))
+                    >> Self::FRACTIONAL_BITS;
+                Self(n as $underlying_type)
+            }
+
+            /// Wrapping (modular) division. Computes self / rhs, wrapping around at the boundary of the type.
+            #[inline]
+            pub fn wrapping_div(self, other: Self) -> Self {
+                let n = (<$intermediate_type>::from(self.0) << Self::FRACTIONAL_BITS).wrapping_div(<$intermediate_type>::from(other.0));
+                Self(n as $underlying_type)
+            }
+
+            /// Wrapping (modular) multiplication.
+            /// Multiplies this fixed-point by an integer, returning the integer result.
+            /// The result will use the full size of the integer. The fractional bits will be truncated.
+            #[inline]
+            pub fn wrapping_mul_int(self, other: $underlying_type) -> $underlying_type {
+                let n = (<$intermediate_type>::from(self.0)
+                    .wrapping_mul(<$intermediate_type>::from(other)))
+                    >> Self::FRACTIONAL_BITS;
+                n as $underlying_type
+            }
+        }
+
+        impl Default for $type_name {
+            /// Returns the default value of `0.0`.
+            #[inline]
+            fn default() -> Self {
+                Self(0)
+            }
+        }
+
+        impl std::fmt::Display for $type_name {
```

Comment:
Would it make sense to `impl` `Debug` differently, to show the raw integer inside?
Well, probably not...

Reply:
It has `#[derive(Debug)]` which does that for us.